module-services/service-test/ServiceTest.cpp

```diff
@@ -3,6 +3,8 @@
 
 #include "service-test/ServiceTest.hpp"
 #include "Timers/TimerFactory.hpp"
+#include "application-test/include/application-test/ApplicationTest.hpp"
```

Comment:
true it's bad. pretty common though :) 
this is this app-specific popup - no other service/app uses this popup - therefore it's not too bad - at least it's stupid simple. 
We have these connected either way as we, in the end, depend on application-common having this function, and application common is dependent to all the apps.
Saying that: it's messed up. But I'll fix it to be messed up the same way in most places it's :)

Reply:
Are you sure its that common ? There are only two includes in services that contain application links and both are leading to application-settings and both are added lately (3months ago and 1 week ago) so we kinda put an effort not to link apps in services.
Also - how is application-common dependent to all apps? :o I thought its reverse - that all apps are dependent on apps-common

Reply:
Sure - let me explain myself :) 
We have cross-references to applications in event manager, application manager (if I recall properly) so in fact in system core. There are other - hidden - in other includes.  And I totally agree that it rather shouldn't take place.

In this particular place, I have to know the application name somehow - because this is the place from which `Launch` is called. just to have this part of code `std::make_unique<app::ApplicationLaunchData>(app::application_test));`

Other apps are rather launched from application desktop so there is no such issue like here.

It could be probably fixed with a separate API library - but we do not have such an approach anywhere. Could be possibly done to extend the manifest - but we do not have it anywhere too.


As for the linking. True - dependency is another way around, my bad. We have specified some direct dependencies, but no explicit dependency app-common to all apps but each app to app-common.

Reply:
great, thanks! I do agree that this dep will have to be here since you need the app name. Maybe write a comment if its a "tutorial" service and we would not like to encourage people to do such things in colaborative code? No pressure though :)